src/gui/owncloudgui.cpp

```diff
@@ -50,34 +51,30 @@ namespace {
 
 using namespace OCC;
 
-QString initLocalFolder()
-{
-    auto localFolder = OCC::Theme::instance()->defaultClientFolder();
-    // Update the local folder - this is not guaranteed to find a good one
-    // if its a relative path, prepend with users home dir, otherwise use as absolute path
-
-    if (!QDir(localFolder).isAbsolute()) {
-        localFolder = QDir::homePath() + QDir::separator() + localFolder;
-    }
-    return OCC::FolderMan::instance()->findGoodPathForNewSyncFolder(localFolder);
-}
-
-void setUpInitialSyncFolder(AccountStatePtr accountStatePtr)
+void setUpInitialSyncFolder(AccountStatePtr accountStatePtr, const QString &localFolder, Wizard::SyncMode syncMode)
 {
     auto account = accountStatePtr->account();
+
     // ensure we are connected and fetch the capabilities
     auto validator = new ConnectionValidator(account, account.data());
-    QObject::connect(validator, &ConnectionValidator::connectionResult, account.data(), [accountStatePtr](ConnectionValidator::Status status, const QStringList &errors) {
+
+    QObject::connect(validator, &ConnectionValidator::connectionResult, account.data(), [accountStatePtr, localFolder, syncMode](ConnectionValidator::Status status, const QStringList &errors) {
```

Comment:
In my book this should be a proper slot method. Too big for a lambda: When you read the surrounding method, you loose context over this huge lambda.